frame/offences/src/mock.rs

```diff
@@ -40,7 +40,6 @@ pub struct OnOffenceHandler;
 
 thread_local! {
 	pub static ON_OFFENCE_PERBILL: RefCell<Vec<Perbill>> = RefCell::new(Default::default());
-	pub static CAN_REPORT: RefCell<bool> = RefCell::new(true);
 	pub static OFFENCE_WEIGHT: RefCell<Weight> = RefCell::new(Default::default());
```

Comment:
I see `set_offence_weight` is removed. Is this needed anymore?

Reply:
No, previously it was used in the `weight_soft_limit_is_used` test which tests the `on_initialize` hook, this hook was submitting deferred offences based on the WeightSoftLimit.

That is actually one concern I have, **DeferredOffences** is an unbounded Vec, previously using the `on_initialize` hook it was not an issue because it was bounded by the **WeightSoftLimit**:
``` rust
fn on_initialize(now: T::BlockNumber) -> Weight {
    let limit = T::WeightSoftLimit::get();
    let mut consumed = Weight::zero();

    <DeferredOffences<T>>::mutate(|deferred| {
	deferred.retain(|(offences, perbill, session)| {
	    if consumed >= limit {
	        true
	    } else {
	        const weight = T::OnOffenceHandler::on_offence(&offences, &perbill, *session);
	        consumed += weight;
	    }
	})
    });

    consumed
}
```

But now we submit everything in the `on_runtime_upgrade` regardless the amount of deferred offences, do we have a high confidence that the `DeferredOffences` will never be big enough to consume more than 2 seconds to process?

Reply:
I think staking is returning true in current release, so on next release there shouldn't be any deferred offences

Also on_offence is returing weight but we don't make use of it, I think the tests doesn't need to tickle this parameter anymore.